server/src-lib/Hasura/GraphQL/Execute/RemoteJoins.hs

```diff
@@ -209,38 +238,38 @@ joinResults :: GQRespValue
             -> [(Batch, EncJSON)]
             -> GQRespValue
 joinResults hasuraValue0 =
-  foldl (uncurry . insertBatchResults) hasuraValue0 . map (fmap f)
+  foldl' (uncurry . insertBatchResults) hasuraValue0 . map (fmap jsonToGQRespVal)
   where
-    f encJson =
-      case OJ.eitherDecode (encJToLBS encJson) >>= parseGQRespValue of
-        Left (fromString-> err) ->
-          appendJoinError emptyResp ("joinResults: eitherDecode: " <> err)
-        Right gqResp@(GQRespValue Nothing _merrors) ->
-          appendJoinError gqResp "could not find join key"
-        Right gqResp -> gqResp
+    jsonToGQRespVal = either mkErr id . parseGQRespValue
+    mkErr = appendJoinError emptyResp . ("joinResults: eitherDecode: " <>) . fromString
 
 emptyResp :: GQRespValue
-emptyResp = GQRespValue Nothing Nothing
+emptyResp = GQRespValue OJ.empty VB.empty
 
 -- | Insert at path, index the value in the larger structure.
 insertBatchResults ::
      GQRespValue
+  -- ^ Original hasura response with accumulated remote responses
   -> Batch
   -> GQRespValue
   -> GQRespValue
-insertBatchResults hasuraResp Batch{..} remoteResp =
-  case inHashmap batchRelFieldPath hasuraData0 remoteData0 of
-    Left err  -> appendJoinError hasuraResp err
-    Right val -> GQRespValue (Just (fst val)) (gqRespErrors hasuraResp)
+insertBatchResults accumResp Batch{..} remoteResp =
```

Comment:
This function is mostly faulty as per reported errors by preview users. Let's think from scratch here.

Reply:
Opened an issue for this here: https://github.com/tirumaraiselvan/graphql-engine/issues/38